services/shhext/api.go

```diff
@@ -418,9 +418,16 @@ func (api *PublicAPI) processPFSMessage(msg *whisper.Message) error {
 
 	handler := EnvelopeSignalHandler{}
 
+	// Notify of added bundles
+	if response != nil && response.AddedBundles != nil {
```

Comment:
Any reason why this logic was moved here? Generally, checking the `err` should be done first and if it's not `nil`, processing other return values should not be continued as it's usually unclear if these values are correct.

If it's possible that `AddedBundles` has a correct value even though `err` is not `nil`, I would think about better interface.

Reply:
> If it's possible that AddedBundles has a correct value even though err is not nil, I would think about better interface.

Correct, that might be the case, is there an idiomatic way to express this?

A few things I can think of:

1) Split the `HandleMessage` in 2 methods, `HandleBundles` & `HandleMessage`, but I feel that breaks encapsulation (i.e. the caller should not care about calling two functions to handle a single message)
2) Pass a function as  a parameter, say `onSucessfulBundleProcessed`
3) change the struct to embeds the errors, but then we would not be using the second argument, which is probably weird
4) Return a slice of errors, with one being the `message` error & one the `bundle` errors

Basically the problem is essentially this one:
https://www.reddit.com/r/golang/comments/58kt6u/returning_multiple_errors_as_a_slice/

I have an sequence of operations to be performed independently, `addingBundles`, `decryptMessage`, but not to break encapsulation I want them to be performed in the same function `handleMessage` and be notified of the success/failure of both.

What do you reckon?

Reply:
I would vote for 2) but would like to hear @adambabik's opinion.

Reply:
> Can chat.ProtocolService have a global handler of new bundles which is passed when it is instantiated? 

I think that should be fine, otherwise I can go for a callback, thanks